src/bin/ion/commands/inspect.rs

```diff
@@ -243,28 +275,36 @@ impl<'a, 'b> IonInspector<'a, 'b> {
                 InspectorAction::LimitReached => break,
             }
 
-            if !is_first_item && !is_last_item {
+            if !is_first_item && !is_last_item && !expr.is_ephemeral() {
                 // If this item is neither the first nor last in the stream, print a row separator.
                 write!(self.output, "{ROW_SEPARATOR}")?;
             }
 
-            match item {
-                SystemStreamItem::SymbolTable(lazy_struct) => {
+            match expr {
+                EExp(eexp) => {
+                    self.inspect_eexp(0, eexp)?;
+                }
+                SymbolTable(lazy_struct) => {
                     self.inspect_symbol_table(lazy_struct)?;
                 }
-                SystemStreamItem::Value(lazy_value) => {
-                    self.inspect_value(0, "", lazy_value, no_comment())?;
+                EncodingDirective(lazy_sexp) => {
+                    self.inspect_value(0, "", lazy_sexp.as_value(), no_comment())?;
+                }
+                Value(lazy_value) => {
+                    if lazy_value.expanded().is_ephemeral() && self.hide_expansion {
+                        // The user has requested that we not show ephemeral values from macros.
+                    } else {
+                        self.inspect_value(0, "", lazy_value, no_comment())?;
+                    }
                 }
-                SystemStreamItem::VersionMarker(marker) => {
+                VersionMarker(marker) => {
                     self.confirm_encoding_is_supported(marker.encoding())?;
                     self.inspect_ivm(marker)?;
                 }
-                SystemStreamItem::EndOfStream(_) => {
+                EndOfStream(end) => {
+                    self.inspect_end_of_stream(end.range().start)?;
```

Comment:
🗺️ The end of the stream now gets a row with an `// End of stream` comment and (more importantly) the offset where the end occurs.